test/cpp/qps/perf_db_client.h

```diff
@@ -0,0 +1,101 @@
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include <iostream>
+#include <memory>
+#include <string>
+#include <cfloat>
+
+#include <grpc/grpc.h>
+#include <grpc++/channel_arguments.h>
+#include <grpc++/channel_interface.h>
+#include <grpc++/client_context.h>
+#include <grpc++/create_channel.h>
+#include <grpc++/credentials.h>
+#include <grpc++/status.h>
+#include "test/cpp/qps/perf_db.grpc.pb.h"
+
+
+namespace grpc{
+namespace testing {
+
+//Manages data sending to performance database server
+class PerfDbClient {
+public:
+  PerfDbClient() {}
+  
+  void init(std::shared_ptr<ChannelInterface> channel) { stub_ = PerfDbTransfer::NewStub(channel); }
+
+  ~PerfDbClient() {}
+
+  //sets the client and server config information
+  void setConfigs(const ClientConfig& clientConfig, const ServerConfig& serverConfig) const;
+  
+  //sets the QPS
+  void setQPS(double QPS) const;
+
+  //sets the QPS per core
+  void setQPSPerCore(double QPSPerCore) const;
+
+  //sets the 50th, 90th, 95th, 99th and 99.9th percentile latency
+  void setLatencies(double percentileLatency50, double percentileLatency90,
+     double percentileLatency95, double percentileLatency99, double percentileLatency99Point9) const;
+
+  //sets the server and client, user and system times
+  void setTimes(double serverSystemTime, double serverUserTime, 
+    double clientSystemTime, double clientUserTime) const;
+
+  //sends the data to the performancew database server
+  int sendData(std::string access_token, std::string test_name, std::string sys_info) const;
+
+private:
+  std::unique_ptr<PerfDbTransfer::Stub> stub_;
+  mutable ClientConfig clientConfig_;
```

Comment:
I think we haven't used the mutable keyword anywhere in grpc. I would advise against starting it. Particularly in this case as it doesn't seem like anything here should actually be mutable. Mutable is supposed to be something that doesn't affect the externally-visible state of the object, and that clearly isn't the case here.